docker/Dockerfile.dev

```diff
@@ -0,0 +1,43 @@
+# This is a container to run the cypress product from cypress.io.
+# It includes openjdk8 because the intention is to be run as a
+# Jenkins process.
+FROM node:4.4.4
+LABEL node-version="4.4.4"
+
+ARG PLUGINS_REPO
+EXPOSE 4200
+WORKDIR /usr/src
+
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64
+
+# Basic dependencies
+RUN echo 'deb http://httpredir.debian.org/debian jessie-backports main' > /etc/apt/sources.list.d/jessie-backports.list
```

Comment:
The mirrors seem to fail consistently. Adding this line before `apt-get upgrade` seemed to do the trick:
https://github.com/rgeissert/http-redirector/issues/74#issuecomment-241682578

```sh
# Insert this line before "RUN apt-get update" to dynamically
# replace httpredir.debian.org with a single working domain
# in attempt to "prevent" the "Error reading from server" error.
RUN sed -i "s/httpredir.debian.org/`curl -s -D - http://httpredir.debian.org/demo/debian/ | awk '/^Link:/ { print $2 }' | sed -e 's@<http://\(.*\)/debian/>;@\1@g'`/" /etc/apt/sources.list
```